packages/blocks/src/api/factory.js

```diff
@@ -93,10 +94,11 @@ export function cloneBlock( block, mergeAttributes = {}, newInnerBlocks ) {
 	return {
 		...block,
 		clientId,
-		attributes: {
-			...block.attributes,
-			...mergeAttributes,
-		},
+		attributes: Object.assign(
+			{},
```

Comment:
Sorry, I may have miscommunicated. This line is not necessary at all, which is the main optimization over what you had previously.

```suggestion
```